pkg/kubelet/cm/topologymanager/fake_topology_manager.go

```diff
@@ -55,3 +55,21 @@ func (m *fakeManager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAd
 		Admit: true,
 	}
 }
+
+type fakeManagerWithHint struct {
+	fakeManager
+	hint *TopologyHint
+}
+
+// NewFakeMangerWithHint returns an instance of FakeManager with specified topology hints
+func NewFakeMangerWithHint(hint *TopologyHint) Manager {
+	return &fakeManagerWithHint{
+		fakeManager: fakeManager{},
+		hint:        hint,
+	}
+}
+
+func (m *fakeManagerWithHint) GetAffinity(podUID string, containerName string) TopologyHint {
+	klog.Infof("[fake topologymanager] GetAffinity podUID: %v container name:  %v", podUID, containerName)
+	return *m.hint
+}
```

Comment:
Why is this added?

This new `fakeManagerWithHint` type doesn't appear to be used anywhere.

Reply:
it used under the future commit that introduces unittest for the static policy

Reply:
I see it now. I did a grep for `NewFakeManagerWithHint` and it didn't find anything.
It's because it is misspelled here.
It should be `NewFakeManagerWithHint`, but it's misspelled as `NewFakeMangerWithHint `.

That said, I don't think we should add this here. If you need it for a test, it should be added there.

This `fakeManager` (which really should be called `stubManager` is meant to stand as a placeholder when the `TopologyManagerFeatureGate` is disabled -- it's not necessarily meant for easing test development.

Reply:
I will move the creation of fake manager under the static policy unittest commit, do you prefer to move this fake manager to the separate file(because like you said it not necessarily meant for easing test development)?

Reply:
Ah it already under the relevant commit

Reply:
Will fix typo

Reply:
I was suggesting to put it directly under `memorymanager/policy_static_test.go` and not as part of the topologymanager code at all. It just seems awkward here since this is really meant to provide a stub for when we don't actually want a `TopologyManager()` (but need some component there as a placeholder).

Reply:
ok will move it

Reply:
done